backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandDAO.java

```diff
@@ -573,6 +572,30 @@ public boolean isAvoidRecreate(final I_C_Invoice_Candidate ic)
 				.addColumn(I_C_Invoice_Candidate.COLUMN_C_Invoice_Candidate_ID);
 
 		return icQueryBuilder
+				.create()
+				.list();//
+	}
+
+	@NonNull
+	public List<I_C_Invoice_Candidate> retrieveInvoiceCandidates(@NonNull final InvoiceId invoiceId)
+	{
+		final I_C_Invoice invoice = load(invoiceId, I_C_Invoice.class);
+
+		final IQuery<I_C_Invoice_Line_Alloc> invoiceLineAllocQuery =
+				queryBL.createQueryBuilder(I_C_InvoiceLine.class, invoice)
+						.addEqualsFilter(I_C_InvoiceLine.COLUMNNAME_C_Invoice_ID, invoice.getC_Invoice_ID())
+						//collect invoice line alloc
+						.andCollectChildren(I_C_Invoice_Line_Alloc.COLUMN_C_InvoiceLine_ID)
+						.addOnlyActiveRecordsFilter()
+						.create();
```

Comment:
it's OK like this but you *might* (didn't check) also use `andCollect` to get from `I_C_Invoice_Line_Alloc` to `I_C_Invoice_Candidate`
for next time maybe..

Reply:
Oh, just checked the method, that's pretty cool, I'll give it a try. thx!